applications/GeoMechanicsApplication/custom_elements/transient_Pw_line_element.h

```diff
@@ -0,0 +1,419 @@
+             ++integration_point_index) {
+            mRetentionLawVector[integration_point_index]->FinalizeSolutionStep(RetentionParameters);
+        }
+    }
+
+
+    array_1d<double, TNumNodes> CalculateFluidBodyVector(
+        const Matrix& r_N_container,
+        const GeometryType::ShapeFunctionsGradientsType& dN_dX_container,
+        const ProcessInfo& rCurrentProcessInfo,
+        const Vector& rIntegrationCoefficients) const
+    {
+        std::size_t number_integration_points = GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
+        GeometryType::JacobiansType J_container;
+        J_container.resize(number_integration_points, false);
+        for (std::size_t i = 0; i < number_integration_points; ++i) {
+            J_container[i].resize(GetGeometry().WorkingSpaceDimension(), GetGeometry().LocalSpaceDimension(), false);
+        }
+        GetGeometry().Jacobian(J_container, this->GetIntegrationMethod());
+
+        const auto& r_properties = GetProperties();
+        BoundedMatrix<double, 1, 1> constitutive_matrix;
+        GeoElementUtilities::FillPermeabilityMatrix(constitutive_matrix, r_properties);
+
+        RetentionLaw::Parameters    RetentionParameters(GetProperties(), rCurrentProcessInfo);
+
+        array_1d<double, TNumNodes * TDim> VolumeAcceleration;
+        GeoElementUtilities::GetNodalVariableVector<TDim, TNumNodes>(VolumeAcceleration, GetGeometry(), VOLUME_ACCELERATION);
+        array_1d<double, TDim> BodyAcceleration;
+
+        array_1d<double, TNumNodes> fluid_body_vector = ZeroVector(TNumNodes);
+        for (unsigned int integration_point_index = 0;
+             integration_point_index < GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
+             ++integration_point_index) {
+            GeoElementUtilities::InterpolateVariableWithComponents<TDim, TNumNodes>(
+                BodyAcceleration, r_N_container, VolumeAcceleration, integration_point_index);
+
+            array_1d<double, TDim> tangent_vector = CalculateTangentialElementUnitVectorOnIntegrationPoint(integration_point_index, J_container);
+            array_1d<double, 1> projected_gravity = ZeroVector(1);
+            projected_gravity(0) = MathUtils<double>::Dot(tangent_vector, BodyAcceleration);
+
+            const auto N = Vector{row(r_N_container, integration_point_index)};
+            const BoundedMatrix<double, TNumNodes, 1> Temp = prod(dN_dX_container[integration_point_index], constitutive_matrix);
+            double RelativePermeability = mRetentionLawVector[integration_point_index]->CalculateRelativePermeability(RetentionParameters);
+            fluid_body_vector += r_properties[DENSITY_WATER] / r_properties[DYNAMIC_VISCOSITY] * RelativePermeability 
+                * prod(Temp, projected_gravity) * rIntegrationCoefficients[integration_point_index];
+        }
+        return fluid_body_vector;
+    }
```

Comment:
Also here, there is some overlap with the CalculateFluidBodyFlow in the upw small strain element, we might be able to extract this into common functionality

Reply:
@rfaasse @markelov208 
I added an issue to the backlog. 
[GeoMechanicsApplication] Remove duplications in thermal and Pw line elements.

Reply:
Indeed, the duplication arises because the new element does not inherit from the UPw element. Consequently, I had to copy functionalities from there. As far as I am aware, both the Pw and UPw elements will undergo refactoring, potentially changing their structure and inheritance mechanisms. Perhaps it would be a good idea to remove such duplications after the refactoring is completed?

Reply:
I think indeed that is a good idea, it's out of scope for this PR